src/portalbackend/PortalBackend.DBAccess/Repositories/ConnectorsRepository.cs

```diff
@@ -137,16 +137,17 @@ public Connector AttachAndModifyConnector(Guid connectorId, Action<Connector>? i
             .SingleOrDefaultAsync();
 
     /// <inheritdoc />
-    public Task<(bool IsValidConnectorId, bool IsProvidingOrHostCompany, Guid? SelfDescriptionDocumentId, DocumentStatusId? DocumentStatusId, ConnectorStatusId ConnectorStatus, IEnumerable<Guid> AssignedOfferSubscriptions)> GetConnectorDeleteDataAsync(Guid connectorId, Guid companyId) =>
+    public Task<DeleteConnectorData?> GetConnectorDeleteDataAsync(Guid connectorId, Guid companyId) =>
         _context.Connectors
             .Where(x => x.Id == connectorId)
-            .Select(connector => new ValueTuple<bool, bool, Guid?, DocumentStatusId?, ConnectorStatusId, IEnumerable<Guid>>(
-                true,
+            .Select(connector => new DeleteConnectorData(
                 connector.ProviderId == companyId || connector.HostId == companyId,
                 connector.SelfDescriptionDocumentId,
                 connector.SelfDescriptionDocument!.DocumentStatusId,
                 connector.StatusId,
-                connector.ConnectorAssignedOfferSubscriptions.Select(x => x.OfferSubscriptionId)
+                connector.ConnectorAssignedOfferSubscriptions.Select(x => new ConnectorOfferSubscription(
+                    x.OfferSubscriptionId, x.OfferSubscription!.OfferSubscriptionStatusId
+                ))
```

Comment:
Please add a linebreak

```suggestion
                connector.ConnectorAssignedOfferSubscriptions.Select(x => new ConnectorOfferSubscription(
                    x.OfferSubscriptionId,
                    x.OfferSubscription!.OfferSubscriptionStatusId
                ))
```